compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DataflowAnalyzer.java

```diff
@@ -579,6 +581,17 @@ public void visit(BLangIndexBasedAccess indexAccessExpr) {
     @Override
     public void visit(BLangInvocation invocationExpr) {
         analyzeNode(invocationExpr.expr, env);
+        if (invocationExpr.expr != null && invocationExpr.expr.getKind() == NodeKind.SIMPLE_VARIABLE_REF
+                && ((BLangSimpleVarRef) invocationExpr.expr).getVariableName().value.equals(Names.SELF.value)) {
```

Comment:
We've only handled self as the var on which the invocation is do IINM? 

This wouldn't handle `self` as an argument to a function right? That also should be disallowed.

Reply:
+1, Changed